pkg/reconciler/instances/istio/tests/integration_test.go

```diff
@@ -0,0 +1,167 @@
+package tests
+
+import (
+	"fmt"
+	"net/http"
+	"os"
+	"testing"
+
+	"github.com/avast/retry-go"
+	"github.com/stretchr/testify/require"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/runtime/serializer/yaml"
+	"k8s.io/kubectl/pkg/util/podutils"
+)
+
+const (
+	istioNamespace         = "istio-system"
+	pilotIngressGwSelector = "istio in (pilot, ingressgateway)"
+	healthzTestURLFormat   = "http://%s:%s/healthz/ready"
+	gatewayManifest        = `
+apiVersion: networking.istio.io/v1beta1
+kind: Gateway
+metadata:
+  name: test-gateway
+spec:
+  selector:
+    istio: ingressgateway # use Istio default gateway implementation
+  servers:
+    - port:
+        number: 80
+        name: http
+        protocol: HTTP
+      hosts:
+        - "*"
+`
+	virutalServiceManifest = `
+apiVersion: networking.istio.io/v1beta1
+kind: VirtualService
+metadata:
+  name: istio-healthz
+spec:
+  gateways:
+  - istio-system/test-gateway
+  hosts:
+  - "*"
+  http:
+  - match:
+    - uri:
+        exact: /healthz/ready
+    route:
+    - destination:
+        host: istio-ingressgateway.istio-system.svc.cluster.local
+        port:
+          number: 15021
+`
+)
+
+var (
+	gatewayGVR = schema.GroupVersionResource{Resource: "gateways", Group: "networking.istio.io", Version: "v1beta1"}
+	vsGVR      = schema.GroupVersionResource{Resource: "virtualservices", Group: "networking.istio.io", Version: "v1beta1"}
+)
+
+func TestIstioIntegration(t *testing.T) {
+	skipTestIfDisabled(t)
+
+	setup := newIstioTest(t)
+	defer setup.contextCancel()
+
+	options := metav1.ListOptions{
+		LabelSelector: "app.kubernetes.io/instance=istio",
+	}
+
+	clientset, err := setup.kubeClient.Clientset()
+	require.NoError(t, err)
+
+	podsList, err := clientset.CoreV1().Pods(namespace).List(setup.context, options)
+	require.NoError(t, err)
+
+	for i, pod := range podsList.Items {
+		setup.logger.Infof("Pod %v is deployed", pod.Name)
+		require.Equal(t, v1.PodPhase("Running"), pod.Status.Phase)
+		ready := podutils.IsPodAvailable(&podsList.Items[i], 0, metav1.Now())
+		require.Equal(t, true, ready)
+	}
+
+	t.Run("istio pods should be running", func(t *testing.T) {
```

Comment:
Imho the test base above verifies that pods are running, specifically because of the used assertion: `require.Equal(t, v1.PodPhase("Running"), pod.Status.Phase)` - shouldn't we rephrase this title then to something else?

Reply:
see comment above